### PR TITLE
rsx: Deal with some static analysis warnings

### PR DESCRIPTION
**Pull request by kd-11**
See https://github.com/RPCS3/rpcs3/issues/6014

### CODE REVIEW

File: rpcs3/Emu/RSX/rsx_utils.h [hunk 1]
```diff
@@ -255,6 +255,11 @@ namespace rsx
 		return static_cast<u32>((1ULL << 32) >> utils::cntlz32(x - 1, true));
 	}
 
+	static inline bool fcmp(float a, float b, float epsilon = 0.000001f)
```
Comment (elad335) [inline rpcs3/Emu/RSX/rsx_utils.h]: https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon

Comment (elad335) [reply rpcs3/Emu/RSX/rsx_utils.h]: and it really shouldnt be an rsx function only, some code in cellVdec can also benefit from this etc.

File: rpcs3/Emu/RSX/rsx_methods.cpp [hunk 2]
```diff
@@ -1062,7 +1062,7 @@ namespace rsx
 				clip_x > 0 || clip_y > 0 ||
 				convert_w != out_w || convert_h != out_h;
 
-			const bool need_convert = out_format != in_format || std::abs(scale_x) != 1.0 || std::abs(scale_y) != 1.0;
+			const bool need_convert = out_format != in_format || !rsx::fcmp(scale_x, 1.f) || !rsx::fcmp(scale_y, 1.f);
```
Comment (elad335) [inline rpcs3/Emu/RSX/rsx_methods.cpp]: missing abs.

File: rpcs3/Emu/RSX/rsx_methods.cpp [hunk 3]
```diff
@@ -2282,14 +2274,6 @@ namespace rsx
 		registers[0xa34 / 4] = 0x45000000;
 		registers[0xa38 / 4] = 0x3f000000;
 		registers[0xa3c / 4] = 0x0;
-		registers[NV4097_SET_VIEWPORT_OFFSET] = 0x45000000;
```
Comment (elad335) [inline rpcs3/Emu/RSX/rsx_methods.cpp]: idk why you remove it, but atleast comment it out because it has documentation purposes.

Comment (kd-11) [reply rpcs3/Emu/RSX/rsx_methods.cpp]: It repeats twice on realhw due to hw bugs. I'll leave a comment.

Comment (kd-11) [reply rpcs3/Emu/RSX/rsx_methods.cpp]: It repeats twice on realhw due to hw bugs. I'll leave a comment.

File: rpcs3/Emu/RSX/rsx_methods.cpp (hunk 2 above)
Comment (kd-11) [reply rpcs3/Emu/RSX/rsx_methods.cpp]: Thanks. I failed to catch that lol

File: rpcs3/Emu/RSX/rsx_utils.h (hunk 1 above)
Comment (kd-11) [reply rpcs3/Emu/RSX/rsx_utils.h]: Honestly real epsilon is too small for this. I have seen arithmetic drift propagate much faster. I guess we can have that for default though. I'll definately refactor into utils if it is needed there.

**Comment by kd-11** [comment]
I'll refactor fcmp for cellVdec in a separate PR.

**Comment by legend800** [comment]
Would less/better logging potentially have a positive impact on performance in some games, albeit minimal I assume?

**Comment by kd-11** [comment]
Yes, but its better to just disable manually for now as everything is still so experimental.

**Comment by arcadesfan** [comment]
Hi, do this include any fix for Fatal Error of PR #6290 or it will be realesed soon in a separate build ?

**Pull request merged**